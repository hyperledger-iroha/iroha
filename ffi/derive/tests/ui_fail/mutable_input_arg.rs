use iroha_ffi::{ffi_bindgen, FfiResult};
use std::mem::MaybeUninit;

struct FfiStruct {
    a: u32,
}

#[ffi_bindgen]
impl FfiStruct {
    pub fn from_mutable_input_arg(a: &mut u32) -> Self {
        let output = Self { a: a.clone() };
        *a = 42;
        output
    }
}

fn main() -> Result<(), ()> {
    let s: MaybeUninit<*mut FfiStruct> = MaybeUninit::uninit();

    if FfiResult::Ok != ffi_struct_from_mutable_input_arg(s.as_mut_ptr()) {
        return Err(());
    }

    Ok(())
}
