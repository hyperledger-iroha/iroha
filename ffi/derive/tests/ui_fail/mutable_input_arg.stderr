error: Mutable references not supported
  --> tests/ui_fail/mutable_input_arg.rs:10:38
   |
10 |     pub fn from_mutable_input_arg(a: &mut u32) -> Self {
   |                                      ^^^^^^^^

error[E0425]: cannot find function `ffi_struct_from_mutable_input_arg` in this scope
  --> tests/ui_fail/mutable_input_arg.rs:20:25
   |
20 |     if FfiResult::Ok != ffi_struct_from_mutable_input_arg(s.as_mut_ptr()) {
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
